src/DotNetWorker.Core/ExtensionStartup/IWorkerExtensionStartup.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Core
+{
+    /// <summary>
+    /// A type representing the contract for worker extension startup implementation.
+    /// </summary>
+    public interface IWorkerExtensionStartup
```

Comment:
Extension authors need to implement this interface if they want to participate in the startup.

Reply:
I wonder if we should make this an abstract class `WorkerExtensionStartup` instead of an interface. We bumped into this with `IWebJobsStartup` when we later wanted to add additional methods to it and wound up with `IWebJobsStartup2` and a bunch of plumbing code to handle that.

An abstract class makes this easier as you can add methods with default implementations in the future.

Reply:
Yea, I always wondered about the `IWebJobsStartup2`. I switched this to an abstract base class now.